qiskit/pulse/schedule.py

```diff
@@ -232,6 +233,15 @@ def channels(self) -> Tuple[Channel]:
         """Returns channels that this schedule uses."""
         return tuple(self._timeslots.keys())
 
+    @property
+    def frames(self) -> Tuple[Frame]:
```

Comment:
A tuple implies ordering, but the set that the return value is generated from is unordered, so the ``.frames`` value can change its ordering over time. IMHO it makes sense to just return the explicitly unordered ``Set[Frame]``.